### PR TITLE
Project creation flow v1

### PR DESCRIPTION
**Pull request by mateodaza**
This version still needs some work on

- [x]  Toasts for error handling
- [x]  Improvements on success view ux/ui
- [x]  Improvements on auth flow (Check if user is able to create a project or not)
- [x]  Map feels a bit clunky
- [x] Project preview functionality

 @MohammadPCh if you can help out while I'm back I'd be really grateful <3

Related to Issue [#32](https://github.com/Giveth/giveth-io-typescript/issues/32)

### CODE REVIEW

**Comment by MohammadPCh** [comment]
Thanks @mateodaza , I hope I can help on this :D

**Comment by MohammadPCh** [comment]
Thanks @RamRamez 
It has a build error. can you please fix it?

File: src/components/views/create/FormValidation.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	const _errors = { ...errors };
+	const stringSize = encodeURI(description).split(/%..|./).length - 1;
+	if (stringSize > 4000000) {
+		_errors[ECreateErrFields.DESCRIPTION] = 'Description is too long';
+		setErrors(_errors);
+	} else {
+		_errors[ECreateErrFields.DESCRIPTION] = '';
+		setErrors(_errors);
+	}
+};
```
Comment (MohammadPCh) [inline src/components/views/create/FormValidation.tsx:107]: I think it's something like a helper and it's not a component. so we can move it.

File: src/components/views/create/ImageInput.tsx [hunk 2]
```diff
@@ -57,11 +57,7 @@ const ImageInput = (props: any) => {
...
 				<Buttons>
-					<Button
-						label='Upload cover image'
-						buttonType='primary'
-						onClick={open}
-					/>
+					{/*<Button label='Upload cover image' buttonType='primary' />*/}
 					{/* <Button
 						label='Search for photos'
 						buttonType='texty'
```
Comment (MohammadPCh) [inline src/components/views/create/ImageInput.tsx:63]: please remove them if we don't need them.

File: src/components/views/create/NameInput.tsx [hunk 3]
```diff
@@ -1,25 +1,62 @@
...
 		<>
-			<H5>Name of your Project</H5>
+			<H5 id={ECreateErrFields.NAME}>Name of your Project</H5>
 			<InputContainer>
 				<Label>Project name</Label>
-				<Input
-					placeholder='My First Project'
-					onChange={e => setValue(e.target.value)}
-				/>
+				<div style={{ position: 'relative' }}>
```
Comment (MohammadPCh) [inline src/components/views/create/NameInput.tsx:26]: please don't use inline styles.

File: src/components/views/create/NameInput.tsx [hunk 4]
```diff
@@ -1,25 +1,62 @@
...
+	setValue: (e: string) => void;
+	error: string;
+}) => {
+	const { setValue, value, error } = props;
+	const [characterLength, setCharacterLength] = useState(0);
+	const maxLength = 55;
 	return (
 		<>
-			<H5>Name of your Project</H5>
+			<H5 id={ECreateErrFields.NAME}>Name of your Project</H5>
```
Comment (MohammadPCh) [inline src/components/views/create/NameInput.tsx:23]: I can't get why you add an id to it.

File: src/components/views/create/NameInput.tsx [hunk 5]
```diff
@@ -1,25 +1,62 @@
...
+	font-weight: 500;
+	border-radius: 64px;
+	width: 52px;
+	height: 30px;
+	position: absolute;
+	right: 16px;
+	top: 0;
+	bottom: 0;
+	margin: auto 0;
+`;
```
Comment (MohammadPCh) [inline src/components/views/create/NameInput.tsx:60]: Do we have a typography component for it?

File: src/components/views/create/WalletAddressInput.tsx [hunk 6]
```diff
@@ -19,20 +35,33 @@ const WalletAddressInput = (props: any) => {
...
 	);
 };
 
-const Input = styled(Regular_Input)``;
+const ChangeAddress = styled.div`
+	color: ${brandColors.pinky[500]};
+	font-size: 14px;
+	margin-top: 16px;
+	cursor: pointer;
+`;
```
Comment (MohammadPCh) [inline src/components/views/create/WalletAddressInput.tsx:64]: If we don't have it on typography ask mo to fix it.

File: src/lib/helpers.ts [hunk 7]
```diff
@@ -84,7 +84,10 @@ export const getGasPreference = (
 
 export const isSSRMode = typeof window === 'undefined';
 
-export const compareAddresses = (add1: string, add2: string) => {
+export const compareAddresses = (
+	add1: string | undefined,
+	add2: string | undefined,
+) => {
```
Comment (MohammadPCh) [inline src/lib/helpers.ts:90]: please fix all usages of compareAddresses

**Review by MohammadPCh — CHANGES_REQUESTED** [review]
Thank you @RamRamez ,
It's a great job. I add some suggestions to your code.

File: src/components/views/create/NameInput.tsx (hunk 4 above)
Comment (RamRamez) [reply src/components/views/create/NameInput.tsx:23]: on publish, we use that ID to scroll to the error

File: src/components/views/create/NameInput.tsx (hunk 5 above)
Comment (RamRamez) [reply src/components/views/create/NameInput.tsx:60]: We can use one styled component for each one

**Comment by mateodaza** [comment]
@RamRamez @MohammadPCh is this ready to go?

**Comment by RamRamez** [comment]
> @RamRamez @MohammadPCh is this ready to go?

No please do not merge. I've fixed the issues that @MohammadPCh mentioned but I haven't sent the PR yet.

File: src/components/views/create/SuccessfulCreation.tsx [hunk 8]
```diff
@@ -65,9 +68,8 @@ const GiverH4 = styled(H4)`
 const BGImg = styled.img`
 	position: absolute;
 `;
-
 const Container = styled.div`
-	background-image: url('/images/creation_success.svg');
+	background-image: url('../../../../public/images/creation_success.svg');
```
Comment (MohammadPCh) [inline src/components/views/create/SuccessfulCreation.tsx:72]: @RamRamez 
It's not the correct way. I think the previous code is correct.

File: src/components/views/create/SuccessfulCreation.tsx [hunk 9]
```diff
@@ -23,15 +25,15 @@ const SuccessfulCreation = (props: IProjectBySlug) => {
...
+					Your project is being reviewed by our team. You&apos;ll
+					receive an email from us once your project is listed.
 				</SuccessMessage>
 				<SocialBox project={project} isSuccess />
 				<SuccessMessage>
 					You can still access your project from your account and
 					share it with your friends.
 				</SuccessMessage>
-				<Link passHref href='/projects'>
+				<Link passHref href={slugToProjectView(project.slug)}>
```
Comment (MohammadPCh) [inline src/components/views/create/SuccessfulCreation.tsx:36]: @RamRamez 
why do you use slugToProjectView?

**Review by MohammadPCh — CHANGES_REQUESTED** [review]

File: src/components/views/create/SuccessfulCreation.tsx (hunk 9 above)
Comment (RamRamez) [reply src/components/views/create/SuccessfulCreation.tsx:36]: since we may change the routing in future

**Review by MohammadPCh — APPROVED** [review]
LGTM, Thank you @RamRamez

**Pull request merged**